python/lsst/daf/butler/formatters/matplotlib.py

```diff
@@ -38,8 +41,22 @@ class MatplotlibFormatter(FileFormatter):
 
     def _readFile(self, path: str, pytype: type[Any] | None = None) -> Any:
         # docstring inherited from FileFormatter._readFile
-        raise NotImplementedError(f"matplotlib figures cannot be read by the butler; path is {path}")
+        return plt.imread(path)
 
     def _writeFile(self, inMemoryDataset: Any) -> None:
         # docstring inherited from FileFormatter._writeFile
         inMemoryDataset.savefig(self.fileDescriptor.location.path)
+
+    @staticmethod
+    def fromArray(cls: np.ndarray) -> plt.Figure:
```

Comment:
This does not need to be a static method in the formatter class. It's okay for this to be a private function in the file that is not exported and only referenced from the yaml file. The first parameter should not be named `cls`, it should be `array` or something.